tests/unit/test__utils.py

```diff
@@ -459,3 +460,89 @@ def side_effect_func(value):
 
     # Run
     _validate_foreign_keys_not_null(metadata, data)
+
+
+@patch('sdv._utils.warnings')
+def test_check_sdv_versions_and_warn_no_missmatch(mock_warnings):
+    """Test that no warnings is raised when no missmatch is produced."""
+    # Setup
+    synthesizer = Mock()
+    synthesizer._fitted_sdv_version = version.public
+    synthesizer._fitted_sdv_enterprise_version = version.enterprise
+
+    # Run
+    check_sdv_versions_and_warn(synthesizer)
+
+    # Assert
+    mock_warnings.warn.assert_not_called()
+
+
+@patch('sdv._utils.warnings')
+def test_check_sdv_versions_and_warn_public_missmatch(mock_warnings):
+    """Test that warnings is raised when public version is missmatched."""
+    # Setup
+    synthesizer = Mock()
+    synthesizer._fitted_sdv_version = '1.0.0'
+    synthesizer._fitted_sdv_enterprise_version = version.enterprise
+
+    # Run
+    check_sdv_versions_and_warn(synthesizer)
```

Comment:
Can we also use pytest logic:
`with pytest.warns(SDVVersionWarning, match=message):`

Reply:
Addressed in [c6051b8](https://github.com/sdv-dev/SDV/pull/1857/commits/c6051b8c59d29769e49fcf8f729673a03e58fe24)